searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp

```diff
@@ -206,6 +211,29 @@ HnswIndex::calc_distance(const TypedCells& lhs, uint32_t rhs_docid) const
     return _distance_func->calc(lhs, rhs);
 }
 
+uint32_t
+HnswIndex::estimate_visited_nodes(uint32_t level, uint32_t doc_id_limit, uint32_t neighbors_to_find, const search::BitVector* filter) const
+{
+    uint32_t m_for_level = max_links_for_level(level);
+    uint64_t base_estimate = uint64_t(m_for_level) * neighbors_to_find + 100;
```

Comment:
Since you are multiplying with neighbors_to_find, will it not be better to use a number closer to average rather than max ?

Reply:
_max_links_for_level()_ returns the configured max for that level. We could divide this by two to get an average (assuming the number of links is evenly distributed at that level).

Reply:
Then I suggest we just keep it as is. Only if we maintained an average, or a number that reflected real world.